File: app/src/main/java/abdullah/elamien/bloodbank/viewmodels/DonateViewModel.java
package abdullah.elamien.bloodbank.viewmodels;


import androidx.lifecycle.ViewModel;

import java.util.Map;

import javax.inject.Inject;

import abdullah.elamien.bloodbank.repository.DonateRepository;

/**
 * Created by AbdullahAtta on 6/19/2019.
 */
public class DonateViewModel extends ViewModel {
    private DonateRepository mRepository;

    @Inject
    public DonateViewModel(DonateRepository mRepository) {
        this.mRepository = mRepository;
    }

    public void applyNewDonor(Map<String, Object> donor, String userId) {
        mRepository.checkIfUserDonatedAndApplyIfNot(userId, donor);
    }

}

File: app/src/main/java/abdullah/elamien/bloodbank/viewmodels/DonateViewModel.kt
package abdullah.elamien.bloodbank.viewmodels


import abdullah.elamien.bloodbank.repository.DonateRepository
import androidx.lifecycle.ViewModel
import org.koin.core.KoinComponent
import org.koin.core.inject

/**
 * Created by AbdullahAtta on 6/19/2019.
 */
class DonateViewModel : ViewModel(), KoinComponent {
    private val mRepository: DonateRepository by inject()

    fun applyNewDonor(donor: Map<String, Any>, userId: String) {
        mRepository.checkIfUserDonatedAndApplyIfNot(userId, donor)
    }

}
